### PR TITLE
added testcafe tests and pr workflow

### PR DESCRIPTION
**Pull request by zoha-jamil**

### CODE REVIEW

File: .github/workflows/pull_request.yaml [hunk 1]
```diff
@@ -21,3 +21,32 @@ jobs:
...
       SLACK_WEBHOOK_URL: ${{ secrets.STAKATER_DELIVERY_SLACK_WEBHOOK }}
+
+  test:
+    runs-on: ubuntu-latest
+    needs: [build]
+    env:
+      IMAGE_PATH: ${{ needs.build.outputs.IMAGE_PATH }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
```
Comment (karl-johan-grahn) [inline .github/workflows/pull_request.yaml:32]: this is in v4

File: testcafe/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+FROM node:16-alpine
```
Comment (karl-johan-grahn) [inline testcafe/Dockerfile:1]: why such an old version of node?

File: testcafe/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+WORKDIR testcafe
+
+ENV XDG_DOWNLOAD_DIR=/opt/JDownloader/Downloads
+
+COPY src src
+COPY package.json package.json
+# COPY .env .
```
Comment (karl-johan-grahn) [inline testcafe/Dockerfile:8]: remove commented code

**Review by karl-johan-grahn — CHANGES_REQUESTED** [review]

File: testcafe/Dockerfile (hunk 2 above)
Comment (zoha-jamil) [reply testcafe/Dockerfile]: I missed this, updated to 22-alpine

File: .github/workflows/pull_request.yaml (hunk 1 above)
Comment (zoha-jamil) [reply .github/workflows/pull_request.yaml]: updated

File: testcafe/Dockerfile (hunk 3 above)
Comment (zoha-jamil) [reply testcafe/Dockerfile]: removed

File: content/managed-addons/monitoring-stack/grafana-dashboard.md [hunk 4]
```diff
@@ -12,139 +12,139 @@ This document explains how to create Grafana Dashboard via GrafanaDashboard CR p
...
-    apiVersion: integreatly.org/v1alpha1
-    kind: GrafanaDashboard
-    metadata:
-      name: grafana-dashboard-example
-      namespace: test-ns
-      labels:
-        grafanaDashboard: grafana-operator
-    spec:
-      json: |-
+```yaml
```
Comment (karl-johan-grahn) [inline content/managed-addons/monitoring-stack/grafana-dashboard.md:15]: I think the indentation needs to be kept as is, otherwise the formatting is incorrect, this is within a numbered list, why did you change it like this?

**Review by karl-johan-grahn — CHANGES_REQUESTED** [review]

File: content/managed-addons/monitoring-stack/grafana-dashboard.md (hunk 4 above)
Comment (zoha-jamil) [reply content/managed-addons/monitoring-stack/grafana-dashboard.md:15]: We can either add yaml by enclosing it in ```yaml ... ``` or by just adding 4 spaces. 
Here both of the methods were applied i.e. adding ```yaml and spaces as well due to which the yaml was invalid and tag (```yaml) was displaying. Removing spaces resolved the issue. 

You can let me know whatever looks right, I can use that method.
1. With indentation (I have added 6 spaces - 4 to add it as yaml and 2 extra spaces for indentation):
Screenshot from 2024-07-18 14-17-40


2. With ```yaml tag
Screenshot from 2024-07-18 14-20-47

Comment (zoha-jamil) [reply content/managed-addons/monitoring-stack/grafana-dashboard.md:15]: And in the first method, we can add more spaces as well

Comment (zoha-jamil) [reply content/managed-addons/monitoring-stack/grafana-dashboard.md:15]: Resolved, looks like this now:
Screenshot from 2024-07-18 14-31-28

Comment (karl-johan-grahn) [reply content/managed-addons/monitoring-stack/grafana-dashboard.md:15]: It's step 2 and so on that you need to make sure really renders

Comment (zoha-jamil) [reply content/managed-addons/monitoring-stack/grafana-dashboard.md]: corrected indentation, it renders correctly now. Screenshots:
Screenshot from 2024-07-19 18-25-15
Screenshot from 2024-07-19 18-25-56

**Review by karl-johan-grahn — APPROVED** [review]
